design/service-binding.md

```diff
@@ -0,0 +1,356 @@
+# Service Binding Proposal for Strimzi
+
+This document is intended to progress discussion about how to enable Strimzi to work well with the Service Binding Operator. It includes some suggested enhancements to both Strimzi and the Service Binding Operator. It would be good to settle on an agreement about how these two technologies can best work together so we can begin the technical work to deliver.
+
+The Service Binding Operator is responsible for binding services such as databases and message brokers to runtime applications in Kubernetes. It's still under development, but the intention is that it becomes part of Operator Lifecycle Management.
+
+## Service Binding Operator today
+
+Today, Strimzi does not fit very nicely with service binding. With the current Service Binding Opeerator implementation, you need to include a template in the `ServiceBindingRequest` to extract the address information from the listener status.
+
+``` yaml
+apiVersion: apps.openshift.io/v1alpha1
+kind: ServiceBindingRequest
+metadata:
+  name: barista-kafka
+spec:
+  applicationSelector:
+    group: apps
+    resource: deployments
+    resourceRef: barista-kafka
+    version: v1
+  backingServiceSelector:
+    group: kafka.strimzi.io
+    kind: Kafka
+    resourceRef: my-cluster
+    version: v1beta1
+  customEnvVar:
+    - name: KAFKA_BOOTSTRAP_SERVERS
+      value: |-
+        {{- range .status.listeners -}}
+          {{- if and (eq .type "plain") (gt (len .addresses) 0) -}}
+            {{- with (index .addresses 0) -}}
+              {{ .host }}:{{ .port }}
```

Comment:
This will be used only within the Kubernetes cluster, right? If that is the case, the address of the bootstrap service is easily identifiable. `<cluster-name>-kafka-botostrap.<namespace>.svc:<port>` - could this be used to simply generate it from some fragments?

Reply:
Service-specific code which depends upon the fragments really belongs in the service. We don't want the Service Binding Operator to understand how Strimzi works or build a dependency on its current naming conventions. Instead, Strimzi could follow the conventions for the SBO using annotations so that the SBO can pluck out the right information itself.